### ISSUE TITLE
Command `pyenv install 3.6.8` fails on Arch ('error: failed to install pip via get-pip.py')

### ISSUE THREAD

**Issue by el-iot** [pyenv/pyenv#2141]
I cannot seem to install python `3.6.8` (I have not tried other versions) on Arch Linux. The last 10 or-so log lines of the command `pyenv install 3.6.8` are:
```
rm -f /home/el/.pyenv/versions/3.6.8/share/man/man1/python3.1
(cd /home/el/.pyenv/versions/3.6.8/share/man/man1; ln -s python3.6.1 python3.1)
if test "xupgrade" != "xno"  ; then \
	case upgrade in \
		upgrade) ensurepip="--upgrade" ;; \
		install|*) ensurepip="" ;; \
	esac; \
	 ./python -E -m ensurepip \
		$ensurepip --root=/ ; \
fi
Looking in links: /tmp/tmp470fs0f0
Collecting setuptools
Collecting pip
Installing collected packages: setuptools, pip
Successfully installed pip-18.1 setuptools-40.6.2
```
Paradoxically, the logs indicate that `pip` could not be installed with `get-pip.py`, yet the final log lines indicate that pip was successfully installed.
Platform / OS:
```
$ uname -rm
5.14.14-arch1-1 x86_64
```
Pyenv:
```
$ pyenv --version
pyenv 2.2.0
```
Python:
```
$ python --version
Python 3.9.7
```
C Compiler information (e.g. gcc 7.3): 
```
$ gcc --version
gcc (GCC) 11.1.0
<copyright statement>
```
Please reproduce the problem with debug tracing enabled and attach the resulting output as a gist
[python-build.20211101142508.2192272.log](https://github.com/pyenv/pyenv/files/7451162/python-build.20211101142508.2192272.log)

Let me know if I have missed anything! I don't often make issues on Github and any feedback would be good :+1:

**Comment by native-api** [comment]
Please attach the console output, not the log file.

**Comment by vakabus** [comment]
I think I might have the same problem. The console output is this:

```
vasek@dingo ~ [1]> pyenv install 3.6.12
Downloading Python-3.6.12.tar.xz...
-> https://www.python.org/ftp/python/3.6.12/Python-3.6.12.tar.xz
Installing Python-3.6.12...


BUILD FAILED (Arch rolling using python-build 20180424)

Inspect or clean up the working tree at /tmp/python-build.20211103110330.314292
Results logged to /tmp/python-build.20211103110330.314292.log

Last 10 log lines:
(cd /home/vasek/.pyenv/versions/3.6.12/share/man/man1; ln -s python3.6.1 python3.1)
if test "xupgrade" != "xno"  ; then \
	case upgrade in \
		upgrade) ensurepip="--upgrade" ;; \
		install|*) ensurepip="" ;; \
	esac; \
	 ./python -E -m ensurepip \
		$ensurepip --root=/ ; \
fi
make: *** [Makefile:1102: install] Error 139
```

The same error occurs also with `3.6.13`. However, `3.7.12` works. I verified, that I have all build dependencies installed and the pyenv version is up to date. Also, before running this, I deleted the `~/.pyenv` directory.

**Comment by native-api** [comment]
@vakabus

> Please reproduce the problem with debug tracing enabled and attach the resulting output as a gist

**Comment by vakabus** [comment]
@native-api Sure. Here it is. Hope I did it correctly.
https://gist.github.com/vakabus/e6eec19cbe40e9091d05e9ef68d90482

**Comment by vakabus** [comment]
```
Wed 2021-11-03 10:24:48 CET   270211 1000 1000 SIGSEGV present  /tmp/python-build.20211103102317.258580/Python-3.6.13/python                                   5.1M
Wed 2021-11-03 10:30:44 CET   284725 1000 1000 SIGSEGV present  /tmp/python-build.20211103102854.272965/Python-3.6.13/python                                   5.1M
Wed 2021-11-03 10:57:17 CET   301387 1000 1000 SIGSEGV present  /tmp/python-build.20211103105532.289646/Python-3.6.13/python                                   5.1M
Wed 2021-11-03 11:02:38 CET   314087 1000 1000 SIGSEGV present  /tmp/python-build.20211103110055.302462/Python-3.6.13/python                                   5.1M
Wed 2021-11-03 11:05:01 CET   326006 1000 1000 SIGSEGV present  /tmp/python-build.20211103110330.314292/Python-3.6.12/python                                   5.0M
Thu 2021-11-04 00:09:22 CET   439827 1000 1000 SIGSEGV present  /home/vasek/.pyenv/versions/3.6.8/bin/python3.6                                                4.8M
Thu 2021-11-04 00:09:24 CET   439841 1000 1000 SIGSEGV present  /home/vasek/.pyenv/versions/3.6.8/bin/python3.6                                                6.5M
```

I guess, it's similar issue with all the minor versions. I've found these coredumps on my system.

With the version 3.6.12, the binary was not deleted and the coredump says, it crashed in the function `PyCFuncPtr_new` in file `Modules/_ctypes/_ctypes.c` (line 3557). Unfortunately, I am not that familiar with Python internals and I am not sure, what it implies or why it could have happened.

**Comment by el-iot** [comment]
@native-api here's my output log (apologies if it I didn't do it correctly)
https://gist.github.com/el-iot/98b1fe095be984673740d37d4d30ab6b

**Comment by native-api** [comment]
@el-iot There are no errors in the attached log.

**Comment by el-iot** [comment]
@native-api Ah right - I don't think I redirected `STDERR` when I dumped the output to a log file (sorry about that). Here's a better one:
https://gist.github.com/el-iot/f65446cac75b1c9e0a55bdbf95f75c04

**Comment by native-api** [comment]
I'm afraid I can't help you with debugging the segfaults.

I don't use Arch; tried to install it to reproduce the problem and get my own ones, and found there's no installer so it's nontrivial. Moreover, it turns out it uses Pacman -- which I've had very bad experience with in MinGW: I've never found a way to get sources or debug symbols. So I've no idea how to debug segfaults in Arch and can't help you here...

Most probably, the particular version of Python is incompatible with some of the dependency libraries. Since Arch is a rolling release, you're pretty much guaranteed to be linking this somewhat old version agains lib versions it was never tested with...

**Comment by achow101** [comment]
#2046 seems to be a similar issue and the solution suggested there (`CFLAGS="-O2"`) resolved this issue for me.

**Comment by LukaK** [comment]
Maybe #1889  will help you. I had trouble installing 3.6.12 on arch and patch worked. Good luck!

**Comment by remanifest** [comment]
I was able to get past this by using clang [per this comment](https://github.com/pyenv/pyenv/issues/2239#issuecomment-1079275184).

`CC=clang pyenv install 3.6.15`

**Comment by sinisa-susnjar** [comment]
Brilliant! This solved the issue for me on WSL / Fedora Remix, thanks!

**Comment by sinisa-susnjar** [comment]
> Brilliant! This solved the issue for me on WSL / Fedora Remix, thanks!



> I was able to get past this by using clang [per this comment](https://github.com/pyenv/pyenv/issues/2239#issuecomment-1079275184).
> 
> `CC=clang pyenv install 3.6.15`

**Comment by sinisa-susnjar** [comment]
> I was able to get past this by using clang [per this comment](https://github.com/pyenv/pyenv/issues/2239#issuecomment-1079275184).
> 
> `CC=clang pyenv install 3.6.15`

Brilliant! This solved the issue for me on WSL / Fedora Remix, thanks!

**Comment by native-api** [comment]
Looks like https://github.com/pyenv/pyenv/pull/2630 is going to solve this.

It backports a patch to make these EOL versions compatible with newer compilers.

**Comment by newco-inuru** [comment]
> #2046 seems to be a similar issue and the solution suggested there (`CFLAGS="-O2"`) resolved this issue for me.

This also solved my issue on WSL/ubuntu 22.04 and pyenv 2.3.22 thaaaaaaanks
